src/apps/Altinn.AccessManagement/src/Altinn.AccessMgmt.Persistence/Services/AssignmentService.cs

```diff
@@ -29,6 +30,136 @@ IDelegationRepository delegationRepository
+    {
+        // Setup filter to be used for fetching client assignments
+        var clientFilter = assignmentRepository.CreateFilterBuilder();
+        clientFilter.Equal(t => t.ToId, toId);
+
+        // Fetch role metadata
+        var roleFilter = roleRepository.CreateFilterBuilder();
+        roleFilter.In(t => t.Code, roles);
+        var roleResult = await roleRepository.Get(roleFilter, cancellationToken: cancellationToken);
+
+        if (roleResult == null || !roleResult.Any() || roleResult.Count() != roles.Length)
+        {
+            throw new ArgumentException($"Filter: {nameof(roles)}, provided contains one or more role identifiers which cannot be found.");
+        }
+
+        var filterRoleIds = roleResult.Select(r => r.Id).ToList();
+
+        // Fetch role-package metadata
+        var rolePackFilter = rolePackageRepository.CreateFilterBuilder();
+        rolePackFilter.In(t => t.RoleId, filterRoleIds);
+        var rolePackageResult = await rolePackageRepository.Get(rolePackFilter, cancellationToken: cancellationToken);
+
+        // Fetch package metadata
+        var packageResult = await packageRepository.Get(cancellationToken: cancellationToken);
+
+        if (!packages.All(p => packageResult.Select(pr => pr.Urn).Contains($"urn:altinn:accesspackage:{p}")))
+        {
+            throw new ArgumentException($"Filter: {nameof(packages)}, provided contains one or more package identifiers which cannot be found.");
+        }
+
+        // Fetch client assignments
+        clientFilter.In(t => t.RoleId, filterRoleIds);
+        var clients = await assignmentRepository.GetExtended(clientFilter, cancellationToken: cancellationToken);
+
+        // Fetch assignment packages
+        QueryResponse<AssignmentPackage> assignmentPackageResult = null;
+        if (roles.Contains(RETTIGHETSHAVER))
+        {
+            var rettighetshaverClients = clients.Where(c => c.RoleId == roleResult.First(r => r.Code == RETTIGHETSHAVER).Id);
+            if (rettighetshaverClients.Any())
+            {
+                var assignmentPackageFilter = assignmentPackageRepository.CreateFilterBuilder();
+                assignmentPackageFilter.In(t => t.AssignmentId, rettighetshaverClients.Select(p => p.Id));
+
+                assignmentPackageResult = await assignmentPackageRepository.Get(assignmentPackageFilter, cancellationToken: cancellationToken);
+            }
+        }
+
+        return BuildConnectionsFromAssignments(clients, assignmentPackageResult, roleResult, packageResult, rolePackageResult, packages);
+    }
+
+    private List<ClientDto> BuildConnectionsFromAssignments(IEnumerable<ExtAssignment> assignments, IEnumerable<AssignmentPackage> assignmentPackages, QueryResponse<Role> roles, QueryResponse<Package> packages, QueryResponse<RolePackage> rolePackages, string[] filterPackages)
```

Comment:
too much :brain:

Reply:
Ja, er litt 🤯 

Operasjonen snur datamodellen fra assignments til unike klienter (from-parties) og populerer tilgangsinfo om hvilke pakker man har gjennom hvilke roller pr. klient.

Så to assignments fra samme party blir til en klient med to sett med tilganger.